File: apps/mobile/lib/features/wishlist/data/dtos/update_wishlist_item_request_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'update_wishlist_item_request_dto.g.dart';

@JsonSerializable(createFactory: false, includeIfNull: false)
class UpdateWishlistItemRequestDto {
  const UpdateWishlistItemRequestDto({
    this.name,
    this.description,
    this.targetPrice,
    this.priority,
    this.status,
    this.url,
    this.imageUrl,
  });

  final String? name;
  final String? description;
  final int? targetPrice;
  final String? priority;
  final String? status;
  final String? url;
  final String? imageUrl;

  Map<String, dynamic> toJson() => _$UpdateWishlistItemRequestDtoToJson(this);
}

File: apps/api/FinanceControl.Shared/Dtos/Request/UpdateWishlistItemRequestDto.cs
using FinanceControl.Shared.Enums;

namespace FinanceControl.Shared.Dtos.Request
{
    public class UpdateWishlistItemRequestDto
    {
        public string? Name { get; set; }
        public string? Description { get; set; }
        public int? TargetPrice { get; set; }
        public EnumWishlistPriority? Priority { get; set; }
        public EnumWishlistStatus? Status { get; set; }
        public string? Url { get; set; }
        public string? ImageUrl { get; set; }
    }
}
